swift/ExampleLibrary/Sources/ExampleLibraryObjC/include/SomeApiObjC.h

```diff
@@ -0,0 +1,22 @@
+#import <Foundation/Foundation.h>
+
+typedef struct {
+    int32_t first;
+    int32_t second;
+} TwoIntegersObjC;
+
+typedef NS_ENUM(uint16_t, ModeObjC) {
+    ModeObjCUnknown,
+    ModeObjCIdle,
+    ModeObjCActive
+};
+
+@interface SomeApiObjC : NSObject
+
+- (void) poke;
+- (nonnull NSString *) echo:(NSString * _Nonnull)send_value;
+- (int32_t) add:(TwoIntegersObjC)two_integers;
+- (int32_t) multiply:(int32_t)first :(int32_t)second;
+- (void) subscribe_mode:(void(^_Nonnull)(ModeObjC))callback;
```

Comment:
I would use camel style here especially as these are public interfaces that will be used in swift:
```
- (nonnull NSString *) echo:(NSString * _Nonnull)sendValue;
- (int32_t) add:(TwoIntegersObjC)twoIntegers;
- (void) subscribeMode:(void(^_Nonnull)(ModeObjC))callback;